libp2p/protocols/identify.nim

```diff
@@ -118,18 +126,25 @@ proc identify*(p: Identify,
   var message = await conn.readLp()
   if len(message) == 0:
     trace "identify: Invalid or empty message received!"
-    raise newException(IdentityInvalidMsgError, 
+    raise newException(IdentityInvalidMsgError,
       "Invalid or empty message received!")
 
   result = decodeMsg(message)
-  trace "Identify for remote peer succeded"
 
-  if remotePeerInfo.peerId.isSome and 
-     result.pubKey.isSome and
-     result.pubKey != remotePeerInfo.peerId.get().publicKey:
-    trace "identify: Peer's remote public key doesn't match"
-    raise newException(IdentityNoMatchError,
-      "Peer's remote public key doesn't match")
+  if remotePeerInfo.peerId.isSome and result.pubKey.isSome:
+    let peer = PeerID.init(result.pubKey.get())
```

Comment:
PeerID's has `==` comparison overload, you don't need conversion to string to compare it.